### DISCUSSION TITLE
How to get a prediction of reaction time from GDDM?

### DISCUSSION THREAD

**Discussion by DianaY-J** [mwshinn/PyDDM#102] [Ask for help] [answered]
Hi lovely Github folks, 

My colleague and I are using PYDDM to built a GDDM with varying drift to fit our saccade selection data, and we tested the model performance by BIC. We like our current progress, but for the next step, we also want to let the model to predict the reaction time and compare it with the real reaction time we have, and see the difference.

I looked up the PYDDM cookbook, but i fail to find anything like a prediction function. But I figure someone must have tried it before, or it should be implemented somewhere, so I'm here to ask for your help, to check how to get the prediction of reaction time from it. 

Your help will be a lot appreciated!!

**Comment by mwshinn** [comment, marked as answer]
The output of the DDM is a distribution of RTs, not a point prediction.  A point-wise prediction will probably not be very good.  If you want to get a point prediction, you could take the mean or the maximum of the distribution.

**Reply by DianaY-J** [reply]
Thank you so much for your reply! 

How to get a distribution of RTs from the GDDM model? If i display the model, it only gives the parameters of the model, not really the distribution of RT, like the pic below shows
![Screenshot 2024-06-24 at 10 22 54](https://github.com/mwshinn/PyDDM/assets/121393800/48aeb229-52ea-438d-8d69-8a3515966108)

**Reply by mwshinn** [reply]
Please read the quickstart guide. The first step ("Hello world") shows you how to get the distribution.

**Reply by DianaY-J** [reply]
Thank you! 

I read it again now, and my problem is that we don't use the sample object but we fit our own data, and mymodel.t_domain() didn't really tell me about the frequency of each distribution, and i checked out the API documentation plotting (https://pyddm.readthedocs.io/en/stable/apidoc/plotting.html) and it's blank. How can I get a prediction plot mean reaction time? 

Sorry for these many questions if they are too easy, and thank you so much beforehand!

**Comment by mwshinn** [comment]
Like I already said, the first step "Hello World" plots the distribution.  There is no sample object here.  mymodel.solve().pdf("correct") gets you the distribution.

**Reply by DianaY-J** [reply]
Thx!
